src/app/pages/storage/volumes/volume-import-wizard/volume-import-wizard.component.ts

```diff
@@ -118,48 +75,17 @@ export class VolumeImportWizardComponent implements WizardConfiguration {
       },
     );
     dialogRef.componentInstance.setDescription(this.translate.instant('Importing Pool...'));
-    dialogRef.componentInstance.setCall('pool.import_pool', [{ guid: value.guid }]);
+    dialogRef.componentInstance.setCall('pool.import_pool', [{ guid: this.formGroup.value.guid }]);
     dialogRef.componentInstance.submit();
     dialogRef.componentInstance.success.pipe(untilDestroyed(this)).subscribe(() => {
       dialogRef.close(false);
-
-      this.loader.open();
-      this.ws.call('pool.dataset.query', [[['pool', '=', this.pool]]])
-        .pipe(
-          tap(() => {
-            this.loader.close();
-          }),
-          untilDestroyed(this),
-        ).subscribe((datasets) => {
-          const hasLockedDataset = datasets.some((dataset) => dataset.encrypted && dataset.locked);
-          if (!hasLockedDataset) {
-            this.modalService.closeSlideIn();
-            this.modalService.refreshTable();
-            return;
-          }
-          this.dialogService.confirm({
-            title: helptext.unlock_dataset_dialog_title,
-            message: helptext.unlock_dataset_dialog_message,
-            hideCheckBox: true,
-            buttonMsg: helptext.unlock_dataset_dialog_button,
-          }).pipe(
-            tap(() => {
-              this.modalService.closeSlideIn();
-              this.modalService.refreshTable();
-            }),
-            filter(Boolean),
-            untilDestroyed(this),
-          ).subscribe(() => {
-            this.router.navigate(['/storage', 'id', this.pool, 'dataset', 'unlock', this.pool]);
-          });
-        }, (err) => {
-          this.modalService.closeSlideIn();
-          this.modalService.refreshTable();
-          new EntityUtils().handleWsError(this, err, this.dialogService);
-        });
+      this.isFormLoading = false;
+      this.slideInService.close();
+      this.modalService.refreshTable();
```

Comment:
We don't need to use this. The `IxSlideInService` has an `onClose$` subject that you can subscribe to in the table component and get notified when the slide in panel closes.

Reply:
I tried the approach you've proposed, but it turns out that the data in the table is updated every time after the sidebar is closed. I believe that this is not entirely correct, because there might not be a new pool import.

So I'd like to keep it as it is.